src/Microsoft.AspNet.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -136,6 +139,11 @@ internal static void AddViewServices(IServiceCollection services)
             // These are stateless so their lifetime isn't really important.
             services.TryAddSingleton<ITempDataDictionaryFactory, TempDataDictionaryFactory>();
             services.TryAddSingleton<SaveTempDataFilter>();
+
+            services.TryAddSingleton(ArrayPool<ViewBufferValue>.Shared);
+            // This manages buffer references for the lifetime of a single request. This must be scoped to prevent
+            // memory leaks.
+            services.TryAddScoped<IViewBufferScope, MemoryPoolViewBufferScope>();
```

Comment:
It won't result in a leak with the default pool, it just gives us more optimal perf.